pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -298,12 +306,52 @@ func desiredLoadBalancerService(ci *operatorv1.IngressController, deploymentRef
 		}
 		// Azure load balancers are not customizable and are set to (2 fail @ 5s interval, 2 healthy)
 		// GCP load balancers are not customizable and are set to (3 fail @ 8s interval, 1 healthy)
+
+		if v, err := shouldUseLocalWithFallback(ci, service); err != nil {
+			return true, service, err
+		} else if v {
+			service.Annotations[localWithFallbackAnnotation] = ""
+		}
 	}
 
 	service.SetOwnerReferences([]metav1.OwnerReference{deploymentRef})
 	return true, service, nil
 }
 
+// shouldUseLocalWithFallback returns a Boolean value indicating whether the
+// local-with-fallback annotation should be set for the given service, and
+// returns an error if the given ingresscontroller has an invalid unsupported
+// config override.
+func shouldUseLocalWithFallback(ic *operatorv1.IngressController, service *corev1.Service) (bool, error) {
+	var result bool
+
+	// By default, use local-with-fallback when using the "Local" external
+	// traffic policy.
+	if service.Spec.ExternalTrafficPolicy == corev1.ServiceExternalTrafficPolicyTypeLocal {
```

Comment:
Should we return false if `service.Spec.ExternalTrafficPolicy` is not `local`?
As is, this logic would let a user use an unsupported config override to set the local-with-fallback annotation when `service.Spec.ExternalTrafficPolicy` is `cluster`, which is a state we shouldn't allow, right?

Reply:
I'm kind of ambivalent about allowing the user to do weird things with an unsupported config override.  I'll go ahead and make the suggested change.

Reply:
Fair enough. Seems like some low-hanging fruit that we might as well patch while we can IMO.